src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -1546,6 +1546,9 @@ class Global(var currentSettings: Settings, reporter0: Reporter)
         if (settings.YstatisticsEnabled && settings.Ystatistics.contains(phase.name))
           printStatisticsFor(phase)
 
+        if (!globalPhase.hasNext)
+          runReporting.warnUnusedSuppressions()
```

Comment:
Is this done unconditionally? In silencer, there was a flag to enable it. I think doing it unconditionally may cause problems in cross compiled code (against multiple scala versions or backends) where a warning is not emitted for all compilation targets.

Reply:
That's a good point. But note that you can filter these `unused @silent` warnings with a `-Wconf` setting (I'll add a `UnusedSilent` category, instead of just using `Other`), so I think it should be fine to report them unconditionally. Note that they are warnings, not errors.

Reply:
I added `-Wunused:silent` in the end